src/mtsoroyematatu.c

```diff
@@ -0,0 +1,785 @@
+
+	if (ThreeInARow(theBlankOX, 0, 1, 4) ||
+        ThreeInARow(theBlankOX, 0, 2, 5) ||
+	    ThreeInARow(theBlankOX, 0, 3, 6) ||
+	    ThreeInARow(theBlankOX, 1, 2, 3) ||
+	    ThreeInARow(theBlankOX, 4, 5, 6))
+		return gStandardGame ? lose : win;
+	//else if (AllFilledIn(theBlankOX))
+	//	return tie;
+
+    return undecided;
+}
+
+/**
+ * @brief Given a position P, GetCanonicalPosition(P) shall return a 
+ * position Q such that P is symmetric to Q and 
+ * GetCanonicalPosition(Q) also returns Q.
+ * 
+ * @note This Q is the "canonical position". For the choice of the 
+ * canonical position, it is recommended to choose the 
+ * lowest-hash-value position symmetric to the input position.
+ * 
+ * Ideally, we maximize the number of positions P such that
+ * GetCanonicalPosition(P) != P.
+ * 
+ * @param position : The position to inspect.
+ */
+POSITION GetCanonicalPosition(POSITION position) {
+    return position;
+}
+
+/*********** END SOLVING FUNCTIONS ***********/
+
+
+
+
+/*********** BEGIN TEXTUI FUNCTIONS ***********/
+
+/**
+ * @brief Print the position in a pretty format, including the
+ * prediction of the game's outcome.
+ * 
+ * @param position   : The position to pretty-print.
+ * @param playerName : The name of the player.
+ * @param usersTurn  : TRUE <==> it's a user's turn.
+ * 
+ * @note See GetPrediction() in src/core/gameplay.h to see how
+ * to print the prediction of the game's outcome.
+ */
+void PrintPosition(POSITION position, STRING playerName, BOOLEAN usersTurn) {
+	BlankOX theBlankOx[BOARDSIZE];
+
+	PositionToBlankOX(position,theBlankOx);
+
+	printf("\n             1              :        %s         \n",
+	       gBlankOXString[(int)theBlankOx[0]]);
+    printf("            /|\\             :       /|\\        \n");
+	printf("LEGEND:    2-3-4    BOARD:  :      %s-%s-%s       \n",
+	       gBlankOXString[(int)theBlankOx[1]],
+	       gBlankOXString[(int)theBlankOx[2]],
+	       gBlankOXString[(int)theBlankOx[3]] );
+    printf("          /  |  \\           :     /  |  \\      \n");
+	printf("         5---6---7          :    %s---%s---%s   %s\n\n",
+	       gBlankOXString[(int)theBlankOx[4]],
+	       gBlankOXString[(int)theBlankOx[5]],
+	       gBlankOXString[(int)theBlankOx[6]],
+	       GetPrediction(position,playerName,usersTurn));
+}
+
+/**
+ * @brief Find out if the player wants to undo, abort, or neither.
+ * If so, return Undo or Abort and don't change `move`.
+ * Otherwise, get the new `move` and fill the pointer up.
+ * 
+ * @param position The position the user is at.
+ * @param move The move to fill with user's move.
+ * @param playerName The name of the player whose turn it is
+ * 
+ * @return One of (Undo, Abort, Continue)
+ */
+USERINPUT GetAndPrintPlayersMove(POSITION position, MOVE *move, STRING playerName) {
+    USERINPUT ret;
+    do {
+        /* List of available moves */
+        // Modify the player's move prompt as you wish
+        printf("%8s's move: ", playerName);
```

Comment:
I think this is where the player's move is prompted, so try editing this to make it properly align with the printed board representation above.